civicrm/custom/ext/gov.nysenate.webintegration/Civi/Api4/Action/WebIntegration/MigrateSurveyData.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Civi\Api4\Action\WebIntegration;
+
+use Civi\Api4\Generic\AbstractAction;
+use Civi\Api4\Generic\Result;
+use PHPUnit\Exception;
+/**
+ * NYSS 16799
+ * cv api4 WebIntegration.migrateSurveyData
+ */
+class MigrateSurveyData extends AbstractAction {
+
+  /**
+   * When specified in the API call, will limit migration and results to
+   * this Activity Record.
+   */
+  protected $activity_id;
+
+
+  /**
+   * @inheritDoc
+   */
+  public function _run(Result $result) {
+    \Civi::log()->info('Running NYSS Migrate Survey Data', []);
+
+    // Get all custom "Survey" groups
+    $customGroups = \Civi\Api4\CustomGroup::get($this->getCheckPermissions())
+      ->addSelect('*')
+      ->addWhere('title', 'REGEXP', '^Survey')
+      //->setLimit(25)
+      ->execute();
+
+    // For each survey, get the custom field values (survey submissions) and
+    // populate the appropriate Activity Details of the matching entity
+    foreach ($customGroups as $customGroup) {
+      \Civi::log()->info('Processing Custom Group: ' . $customGroup['title'], [
+        'name' => $customGroup['name'],
+        'id' => $customGroup['id'],
+        'title' => $customGroup['title'],
+        'table_name' => $customGroup['table_name']
+      ]);
+      /*\Civi::log()->debug('Processing Custom Group: ' . $customGroup['title'], [
+        'data' => $customGroup
+      ]);*/
+
+      // find all activities related to this custom group / survey
+      $webform_id = substr($customGroup['name'],7);
+      $webform_name = 'Survey_'.$webform_id;
+      $query = \Civi\Api4\Activity::get($this->getCheckPermissions())
+        ->addSelect($webform_name.'.*', 'Website_Survey.*', '*')
+        ->addWhere('activity_type_id', '=', 90)
```

Comment:
let's reference by the machine name rather than id:
->addWhere('activity_type_id:name', '=', 'Website Survey')